### ISSUE TITLE
disbale dark mode doesn't work

### ISSUE THREAD

**Issue by orenmm** [tomaszpoliszuk/DarkMode-Flipswitch#1]
Switch only enables the dark mode, when pressing the toggle again, light mode doesn't work.

**Comment by tomaszpoliszuk** [comment]
Try to remove (not reinstall) this flipswitch and then install it again.
Let me know if it helped.

**Comment by orenmm** [comment]
The thing is it's now part of the Flipswitch package,
I can remove it, but when I install it it says there is a conflict in the
package and I have to remove both this and activator just to test it.

**Comment by tomaszpoliszuk** [comment]
Yes - if you have flipswitch equal or higher than 1.0.16-beta6 then you shouldn't have this one installed since it's already build in into 1.0.16-beta6.
Please remove this one and proceed into installing flipswitch 1.0.16~rc1.

**Comment by orenmm** [comment]
Removed the tweaks, installed flipswitch 1.0.16~rc1, latest
DarkMode-Flipswitch,
flipconvert.
still acts in the same way, can only enable dark mode, not disabling it.

**Comment by tomaszpoliszuk** [comment]
If you have flipswitch 1.0.16~rc1 then uninstall DarkMode-Flipswitch.

**Comment by orenmm** [comment]
sorry! i wrote the wrong version, meant to say: flipswitch 1.0.16 beta 5,
that does not include DarkMode-Flipswitch.
I get the same result with this option and with flipswitch 1.0.16~rc1
installed (automatically removes DarkMode-Flipswitch).

**Comment by tomaszpoliszuk** [comment]
Ok, labeled this as bug.
I'll soon try to replicate it on my device and ask more questions if necessary.

**Comment by shtuseriff** [comment]
Just wanted to report that I have the same issue. I can’t disable dark mode from the switch; only enable it. 
Flipswitch v1.0.16~rc1. 

PS, thanks for making this. I will FINALLY be able to use Flipconvert and theme my damn cc toggles. Held off because there was no DarkMode FS, and therefore it couldn’t be themed. Thank u ❤️

**Comment by tomaszpoliszuk** [comment]
@orenmm sorry - I wrote incorrectly - it's build in since beta7 not beta6.

**Comment by orenmm** [comment]
Yeah I checked the "change history" of Flipswitch to know which version to
download.
But both, your standalone version, and the one with it built, are not able
to disable dark mode and return to light mode.

**Comment by tomaszpoliszuk** [comment]
@orenmm @shtuseriff 
I've tested both options:
- Activator 1.9.13-rc3 + Flipswitch 1.0.16-rc1
- Activator 1.9.13-beta6 + Flipswitch 1.0.16-beta6 + DarkMode-Flipswitch 0.2.2

but I recommend to use first one since it have implemented bug fixes and some cool updates.

Assigned "Dark Mode" from "Switches" [not from "Switches (Activate)" - pointing this out "just in case" since Ii have only parts of informations I need] group and i've triggered it from few different events (like touching Touch ID or toggling from custom menu).

In both cases I could turn on and of Dark Mode.

I need you to do this steps:

- remove any Dark Mode flipswitch assignment you have and respring device
- in iCleaner App (I can provide more detailed answer later if you don't know how to use or where to find iCleaner) go to "Cydia Substrate Addons"  and disable everything
- make sure that "Automatically manage dependencies" is enabled
- enable only Activator
- Apply
- Respring
- test Dark Mode flipswitch

Let me know how it works when you do this.

**Comment by orenmm** [comment]
I followed the steps you mentioned. it does work when assign a toggle gesture in **Activator**,
but not as **Flipswitch** in the **ControlCenter**. Thats what me and @shtuseriff were having difficulty to achieve. There is the native iOS toggle, but you can't theme that icon because its animated. it can be themable with your toggle, but we can't disable the dark mode as I stated before. I hope its more clear now

**Comment by tomaszpoliszuk** [comment]
Oh - i didn't know that.

Could you provide me with name of tweak you use to enable usage of flipswitches in Control Center?

**Comment by orenmm** [comment]
CCModules + FlipConvert
 _FlipConvert simply Convert Flipswitch Toggles for CCSupport_

**Comment by tomaszpoliszuk** [comment]
Ok, now i get it.

After test I can confirm - this flipswitch after converting by FlipConvert doesn't work correctly.

However i'm not yet sure if this is something that can/should be corrected on my end or by julioverne.

I'll look more into it in near future 🙂

**Comment by orenmm** [comment]
I remember having an issue with caller ID toggle. The person who was behind that toggle had to fix it on his end.  Cheers :)

**Comment by shtuseriff** [comment]
yes, @orenmm described the issue perfectly. I have _a guess _that it’s not an issue with FlipConvert. I believe I’ve seen similar issues in the past with a different switch and it was the switch itself.
If it helps at all, I should mention that I do **not** have CCModules installed; so its not that. But like Oren, I do have FlipConvert.

**Comment by tomaszpoliszuk** [comment]
Just to be sure - I'm not suggesting anything at that point but since i'm stuck (but I have few ideas how to change it) at making control center toggle (one of my private, still not finished projects) and have some problems with dark mode in another one (another one not yet finished) I'm suspecting that figuring it out should help also in this case.

I'll probably try to investigate it tomorrow a little bit.

I have now FlipConvert installed for tests 🙂

**Comment by shtuseriff** [comment]
Any progress on this? We're so close to a fully themed CC... just need that little Dark Mode switch to play nicely when using FlipConvert. #MakeMyCCGreatAgain <3
Thanks man!

**Comment by tomaszpoliszuk** [comment]
I've made some progress but we are not there yet.
